### PR TITLE
feat(docs): add scroll localStorage to side drawer

### PR DESCRIPTION
**Pull request by rinaok**
On Daily, @olaf-k mentioned the scrolling problem. 
[This is a solution I proposed not long ago](https://github.com/Vonage/vivid-3/pull/603/commits/39024ca8123702cfd874ea6f309868f7b38b21b5), Until we find a third party library for that, I think this can be the solution.
What do you think? @yinonov

### CODE REVIEW

**Comment by yinonov** [comment]
> On Daily, @olaf-k mentioned the scrolling problem. [This is a solution I proposed not long ago](https://github.com/Vonage/vivid-3/pull/603/commits/39024ca8123702cfd874ea6f309868f7b38b21b5), Until we find a third party library for that, I think this can be the solution. What do you think? @yinonov

should it work on this PR recent deployment? it doesn't

**Comment by rinaok** [comment]
> > On Daily, @olaf-k mentioned the scrolling problem. [This is a solution I proposed not long ago](https://github.com/Vonage/vivid-3/pull/603/commits/39024ca8123702cfd874ea6f309868f7b38b21b5), Until we find a third party library for that, I think this can be the solution. What do you think? @yinonov
> 
> should it work on this PR recent deployment? it doesn't

Yeah I know, it works locally but not in the deployment preview

File: apps/docs/assets/scripts/scroll-handler.js [hunk 1]
```diff
@@ -11,13 +11,28 @@ const onScroll = () => {
...
+	}
+}
+
+const setScrollFromLocalStorage = () => {
+	// set sideDrawer's scroll from localStorage
+	let scrollTop = localStorage.getItem("scroll");
+	if (scrollTop) {
+		setTimeout(() => {
+			getAsideElement().scrollTop = scrollTop;
+		}, 1);
```
Comment (olaf-k) [inline apps/docs/assets/scripts/scroll-handler.js:26]: the only way I found to remove flakiness was to wait for the `aside` to be painted and have an actual height
```js
const setScrollFromLocalStorage = () => {
	const asideElement = getAsideElement();
	if (asideElement.offsetHeight > 0)
		asideElement.scrollTop = localStorage.getItem("scroll") ?? 0;
	else
		requestAnimationFrame(setScrollFromLocalStorage);
}
```

**Review by olaf-k — CHANGES_REQUESTED** [review]

**Comment by yinonov** [comment]
> > > On Daily, @olaf-k mentioned the scrolling problem. [This is a solution I proposed not long ago](https://github.com/Vonage/vivid-3/pull/603/commits/39024ca8123702cfd874ea6f309868f7b38b21b5), Until we find a third party library for that, I think this can be the solution. What do you think? @yinonov
> 
> > 
> 
> > should it work on this PR recent deployment? it doesn't
> 
> 
> 
> Yeah I know, it works locally but not in the deployment preview 

Sounds like a race condition

https://github.com/Vonage/vivid-3/pull/813#discussion_r1019344679

File: apps/docs/assets/scripts/scroll-handler.js (hunk 1 above)
Comment (rinaok) [reply apps/docs/assets/scripts/scroll-handler.js]: @olaf-k thank you!!!

File: apps/docs/assets/scripts/scroll-handler.js [hunk 2]
```diff
@@ -11,13 +11,28 @@ const onScroll = () => {
...
 }
 
 (() => {
 	// hook window scroll
 	window.addEventListener('scroll', onScroll);
 
 	customElements.whenDefined('vwc-side-drawer').then(() => {
-		getAsideElement().addEventListener('scroll', onScroll)
+		setScrollFromLocalStorage();
+		getAsideElement().addEventListener('scroll', onScroll);
```
[the remark this answers is not in the document]
Comment (rinaok) [reply apps/docs/assets/scripts/scroll-handler.js:37]: @olaf-k you mean outside of the whenDefined?

[the remark this answers is not in the document]
Comment (olaf-k) [reply apps/docs/assets/scripts/scroll-handler.js:37]: no, I thought the handler did different things for window and aside, don't mind my comment.

**Review by olaf-k — APPROVED** [review]

**Pull request merged**